src/bin/hal/cmd/psbt.rs

```diff
@@ -429,3 +432,106 @@ fn exec_merge<'a>(matches: &clap::ArgMatches<'a>) {
 		print!("{}", base64::encode(&merged_raw));
 	}
 }
+
+fn cmd_rawsign<'a>() -> clap::App<'a, 'a> {
+	cmd::subcommand("rawsign", "sign a psbt with private key and add sig to partial sigs").args(&[
+		cmd::arg("psbt", "PSBT to finalize, either base64/hex or a file path").required(true),
+		cmd::arg("input-idx", "the input index to edit").required(true),
+		cmd::arg("priv-key", "the private key in WIF/hex").required(true),
+		cmd::arg("compressed", "Whether the corresponding pk is compressed")
+			.required(false)
+			.default_value("true"),
+		cmd::opt("raw-stdout", "output the raw bytes of the result to stdout")
+			.short("r")
+			.required(false),
+		cmd::opt("output", "where to save the resulting PSBT file -- in place if omitted")
+			.short("o")
+			.takes_value(true)
+			.required(false),
+	])
+}
+
+// Get the scriptpubkey/amount for the psbt input
+fn get_spk_amt(psbt: &psbt::PartiallySignedTransaction, index: usize) -> (&bitcoin::Script, u64) {
+    let script_pubkey;
+    let amt;
+	let inp = &psbt.inputs[index];
```

Comment:
There is no check before if `index` is within the range of inputs, so perhaps here do
```rust
let inp = psbt.inputs.get(index).expect("input index out of range");
```

Reply:
Hmm, strike this, you do the index again later, so the supermethod should probably check the index.